symfony/assets/scripts/class/navbar/Navbar.ts

```diff
@@ -0,0 +1,191 @@
+import DesktopLayoutStrategy from "./DesktopLayoutStrategy";
+import LayoutStrategy, { LayoutStrategyEvent } from "./LayoutStrategy";
+import MobileLayoutStrategy from "./MobileLayoutStrategy";
+import NavbarElement from "./NavbarElement";
+import NavbarLink, { NavbarLinkPriority } from "./NavbarLink";
+import NavbarMenuButton from "./NavbarMenuButton";
+import Menu from "./Menu";
+
+const PRIORITY_NUMBER: Record<NavbarLinkPriority, number> = {
+  high: 0,
+  medium: 1,
+  low: 2,
+};
+
+/**
+ * Below this number, the navbar will display in the mobile version. Above this
+ * number, the navbar will display in the desktop version.
+ * Number was chosen to target conventional mobile devices as being "mobile" but
+ * is otherwise arbitrary and can be changed as needed.
+ */
+const MOBILE_LAYOUT_BREAKPOINT_PX = 500;
```

Comment:
Optional/FYI: You could consider getting inspired by https://getbootstrap.com/docs/5.3/layout/breakpoints/#available-breakpoints .